packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -185,27 +185,20 @@ class FlutterPlugin implements Plugin<Project> {
 
         File pluginsFile = new File(project.projectDir.parentFile.parentFile, '.flutter-plugins')
         Properties plugins = readPropertiesIfExist(pluginsFile)
-
-        plugins.each { name, _ ->
-            def pluginProject = project.rootProject.findProject(":$name")
-            if (pluginProject != null) {
-                project.dependencies {
-                    if (project.getConfigurations().findByName("implementation")) {
-                        implementation pluginProject
-                    } else {
-                        compile pluginProject
+        // Plugins right now only have a notion of debug or release.
+        // Profile goes to release.
+        def Boolean isDebug = project.gradle.startParameter.taskRequests.any {it.args[0].contains("Debug")}
+        def String mode = isDebug ? "debug" : "release"
+        plugins.each { name, path ->
+            def File pluginDepsFile = new File(Paths.get(path, 'android', 'build', 'deps.txt').toString())
+            project.dependencies {
+                if (pluginDepsFile.exists()) {
+                    def lines = pluginDepsFile.readLines()
+                    for (line in lines) {
+                        api line
```

Comment:
Can we gain some clarity and encapsulation by moving the understanding of `deps.txt` to a new class?  Could we have a PluginDependencies class that knows where to find this file, how to parse the file, and how to apply it to the project?